kalite/distributed/static/js/distributed/rating/views.js

```diff
@@ -0,0 +1,263 @@
+var Backbone = require("../base/backbone");
+var $ = require("../base/jQuery");
+var _ = require("underscore");
+var BaseView = require("../base/baseview");
+var Handlebars = require("../base/handlebars");
+
+/*
+    Container view for feedback form.
+    See the design document here: https://docs.google.com/a/learningequality.org/document/d/1W_2vv1cZU5wZp_MLjvPy6jHwkRVPW_iTJhHEpSOYrCs/edit?usp=drive_web
+*/
+module.exports = BaseView.extend({
+
+    template: require("./hbtemplates/rating.handlebars"),
+
+    events: {
+        "click .rating-submit": "submit_rating",
+        "click .rating-edit": "edit_callback",
+        "click .rating-delete": "delete_callback"
+    },
+
+    initialize: function(options) {
+        /*
+            Prepare self and subviews.
+        */
+        this.model = options.model || function() {
+            var model = new Backbone.Model();
+            model.url = "/api/not/gonna/work";
+            return model;
+        }();
+        _.bindAll(this, "delete_callback", "submit_rating", "edit_callback", "render1st", "render2nd", "render3rd", "renderTextArea", "renderAll", "render");
+    },
+
+    render: function() {
+        if( this.model.isNew() || this.model.get("rating1") === 0) {
+            this.render1st();
+        } else {
+            this.renderAll();
+        }
+    },
+
+    render1st: function() {
+        /*
+            Render the first star rating, then wait until it's been rated to show the next step.
+            Called when the view's model is not fetched successfully -- particularly when it doesn't yet exist.
+        */
+        this.$el.html(this.template());
+        this.$(".rating-submit").hide();
+        this.$(".rating-edit").hide();
+        this.$(".rating-skip").hide();
+        this.$(".rating-delete").hide();
+        this.star_view_1 = this.add_subview(StarView, {
+            el: this.$("#star-container-1"),
+            model: this.model,
+            rating_attr: "rating1"
+        });
+        var self = this;
+        this.model.once("change:rating1", function() {
+            self.star_view_1.remove();
+            self.render2nd();
+        })
+    },
+
+    render2nd: function() {
+        this.star_view_2 = this.add_subview(StarView, {
+            el: this.$("#star-container-2"),
+            model: this.model,
+            rating_attr: "rating2"
+        });
+        var self = this;
+        this.model.once("change:rating2", function() {
+            self.star_view_2.remove();
+            self.render3rd();
+        })
+    },
+
+    render3rd: function() {
+        this.star_view_3 = this.add_subview(StarView, {
+            el: this.$("#star-container-3"),
+            model: this.model,
+            rating_attr: "rating3"
+        });
+        var self = this;
+        this.model.once("change:rating3", function() {
+            self.star_view_3.remove();
+            self.renderTextArea();
+        })
+    },

```

Comment:
Certainly adding a subview & adding the attribute change callback could be spun-off.